### PR TITLE
Fix streamsocket

### LINKED ISSUE #162: Scapy StreamSocket.sr1 problem on OSX [Bug]
system: OSX 10.11.4

When i use StreamSocket.sr1 i got this error.
```python
>>> from scapy.all import *
WARNING: No route found for IPv6 destination :: (no default route?)
>>> s=socket.socket()
>>> s.connect(('127.0.0.1',80))
>>> connection = StreamSocket(s, Raw)
>>> connection.sr1(Raw("Get"))
Begin emission:
Finished to send 1 packets.
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "scapy/supersocket.py", line 52, in sr1
    a,b = sendrecv.sndrcv(self, *args, **kargs)
  File "scapy/sendrecv.py", line 129, in sndrcv
    r = pks.nonblock_recv()
AttributeError: 'StreamSocket' object has no attribute 'nonblock_recv'
```

### PR DESCRIPTION
**Pull request by dark-lbp**
This fixes #162

### CODE REVIEW

File: scapy/sendrecv.py [hunk 1]
```diff
@@ -123,7 +123,7 @@ def sndrcv(pks, pkt, timeout = None, inter = 0, verbose=None, chainCC=0, retry=0
                                 if remaintime <= 0:
                                     break
                             r = None
-                            if arch.FREEBSD or arch.DARWIN:
+                            if not (pks.__class__.__name__ == 'StreamSocket') and (arch.FREEBSD or arch.DARWIN):
```
Comment (p-l-) [inline scapy/sendrecv.py]: You should use `isinstance()` instead of comparing the class name to a string.

Comment (dark-lbp) [inline scapy/sendrecv.py]: I tried and got error. I post the error at #162

Comment (p-l-) [inline scapy/sendrecv.py]: You need to import `StreamSocket` if you want to use it.

**Comment by p-l-** [comment]
Also, please rebase instead of pushing a merge commit.

**Comment by dark-lbp** [comment]
@p-l- Thanks, will close this and submit a new PR.

**Pull request closed without merging**